config/src/main/java/org/springframework/security/config/annotation/web/builders/WebSecurity.java

```diff
@@ -306,6 +315,28 @@ protected Filter performBuild() throws Exception {
 		return result;
 	}
 
+	private void buildPrivilegeEvaluator(List<SecurityFilterChain> securityFilterChains) {
+		if (this.privilegeEvaluator != null || securityFilterChains.size() <= 1) {
+			return;
+		}
+		List<RequestMatcherDelegatingWebInvocationPrivilegeEvaluator.DelegatePrivilegeEvaluator> delegates = new ArrayList<>();
+		for (SecurityFilterChain securityFilterChain : securityFilterChains) {
+			RequestMatcherDelegatingWebInvocationPrivilegeEvaluator.Builder builder = RequestMatcherDelegatingWebInvocationPrivilegeEvaluator
+					.withRequestMatcher(securityFilterChain::matches);
+			for (Filter filter : securityFilterChain.getFilters()) {
+				if (filter instanceof FilterSecurityInterceptor fsi) {
+					builder.withFilterSecurityInterceptor(fsi, this.servletContext);
+					continue;
+				}
+				if (filter instanceof AuthorizationFilter af) {
+					builder.withAuthorizationManager(af.getAuthorizationManager());
+				}
+			}
+			delegates.add(builder.build());
+		}
+		this.privilegeEvaluator = new RequestMatcherDelegatingWebInvocationPrivilegeEvaluator(delegates);
```

Comment:
instead of initializing privilegeEvaluator it should create an instance and be assigned in getPrivilegeEvaluator. See https://github.com/spring-projects/spring-security/pull/10575/files#diff-2a3768a3ea186b6702859073cc35bc02aa6c8ebf6b20f19a67323f7c20d3bd38R295